Gnomad4

Nitram36 opened the pull request:
Rearranged branch so that VRS ID module would not also be submitted. Added data folder, with script that can be used to generate gnomad4.sqlite datbase. Altered python script so that module works when sqlite database is present in data folder. Removed pypi dependency from yml file as instructed.

Nitram36 committed:
Created gnomad4 module.

Create gnomad4 module in a new branch.

Nitram36 committed:
Updated files.

Updated python script and yml file.

Nitram36 committed:
Added folder

Added folder for generating gnomad4 sqlite database.

Nitram36 committed:
Modified gnomad4 folder and deleted vrs_id on gnomad4 branch

Nitram36 committed:
Removed pypi dependency from yml file.

rkimoakbioinformatics commented:
Thanks. `annotators/vrs_id` has been updated by me. Please do the following to make this PR up to date with `master`.

>git checkout master
>git pull
>git checkout gnomad4
>git merge master
>git push

Also, this PR seems not to have the make script for gnomad4.sqlite. Please send the make script to me on Discord.

rkimoakbioinformatics commented:
@Nitram36 Please resolve the conflicts with the master branch. See https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/addressing-merge-conflicts/resolving-a-merge-conflict-on-github.

Nitram36 committed:
Merged vrs_id module from master into gnomad4 to resolve conflicts.

Nitram36 committed:
Merge remote-tracking branch 'upstream/master' into gnomad4

rkimoakbioinformatics commented:
The values seem to be different from gnomAD v4.1.0 website. Can you take a look into the following cases, for example?

From gnomad4.sqlite:
│ CHROM │  POS   │ REF │ ALT │        AF_ALL        │ AF_AFR │ AF_AMR │ AF_ASJ │ AF_EAS │ AF_SAS │ AF_FIN │ AF_MID │        AF_NFE        │
├───────┼────────┼─────┼─────┼──────────────────────┼────────┼────────┼────────┼────────┼────────┼────────┼────────┼──────────────────────┤
│ chr1  │ 925942 │ A   │ G   │ 1.59616001838003e-06 │ 0.0    │ 0.0    │ 0.0    │ 0.0    │ 0.0    │ 0.0    │ 0.0    │ 2.85720989268157e-06 │
│ chr1  │ 925942 │ A   │ T   │ 1.59616001838003e-06 │ 0.0    │ 0.0    │ 0.0    │ 0.0    │ 0.0    │ 0.0    │ 0.0    │ 2.85720989268157e-06 │
(compare to https://gnomad.broadinstitute.org/variant/1-925942-A-T?dataset=gnomad_r4)

From gnomad4.sqlite:
│ CHROM │   POS    │ REF │ ALT │        AF_ALL        │       AF_AFR       │        AF_AMR        │ AF_ASJ │ AF_EAS │        AF_SAS        │        AF_FIN        │ AF_MID │        AF_NFE        │
├───────┼──────────┼─────┼─────┼──────────────────────┼────────────────────┼──────────────────────┼────────┼────────┼──────────────────────┼──────────────────────┼────────┼──────────────────────┤
│ chr1  │ 12666560 │ G   │ A   │ 2.39417004195275e-05 │ 0.0001194740034407 │ 0.000111797002318781 │ 0.0    │ 0.0    │ 2.31862995860865e-05 │ 5.61671986361034e-05 │ 0.0    │ 1.70862003869843e-05 │
(compare to https://gnomad.broadinstitute.org/variant/1-12666560-G-A?dataset=gnomad_r4)

Nitram36 committed:
Added scripts to generate gnomad v4.1 allele frequency database.

Nitram36 commented:
Hi Rick, I see the problem. When I started the project v4.0 was the latest gnomad release. Therefore, the allele frequency database was generated from v4.0. Now, v4.1 has been released and so the allele frequencies have been updated. I have updated the script for generating the database and added it to the module directory. This should now add the updated allele frequencies.
